### PR TITLE
feat(ui5-tooling-modules): improved dependency resolution

### PR DESCRIPTION
**Pull request by petermuessig**
The chunks needed by third-party packages are resolved against the root namespace (/resources) which leads to issues in case of running applications against CDN. Therefore, the chunks are now prepending the namespace of the application to ensure to resolve relative against the application.

For some third-party packages like punycode built-in modules in node exist. In this case, the dependency resolution looked up the node built-in module and the node polyfills have been used instead of the project dependency. This has been changed that for entry points (dependencies who have been declared in any module) are always first resolved against the local dependencies before the polyfills are used.

Fixes #1032

### CODE REVIEW

**Comment by petermuessig** [comment]
@hschaefer123: this feature fixes the issue with the chunks which cannot be resolved - now always the project namespace is prepended to the module names in application projects to ensure a project relative resolution.

@igzThomasFrischholz: this feature fixes the resolution of punycode - sorry the change became a bit more complex after understanding the details of the issue so that the node polyfill had been detected instead of the dependency.

File: packages/ui5-tooling-modules/lib/task.js [hunk 1]
```diff
@@ -91,7 +93,8 @@
 	const removeScopePrefix = config?.removeScopePrefix || config?.removeScopePreceder;
 	// eslint-disable-next-line jsdoc/require-jsdoc
 	function rewriteDep(dep, bundledResources, useDottedNamespace) {
-		const aDep = dep.replaceAll(/\.?\.\//g, "");
+		// remove the relative path and the project namespace
+		const aDep = dep.replaceAll(`${options.projectNamespace}/resources/`, "").replaceAll(/\.?\.\//g, "");
```
[the remark this answers is not in the document]
Comment (petermuessig) [reply packages/ui5-tooling-modules/lib/task.js:97]: Acceptable

**Comment by petermuessig** [comment]
I'll release a bit later today.

**Comment by igzThomasFrischholz** [comment]
> @hschaefer123: this feature fixes the issue with the chunks which cannot be resolved - now always the project namespace is prepended to the module names in application projects to ensure a project relative resolution.
> 
> @igzThomasFrischholz: this feature fixes the resolution of punycode - sorry the change became a bit more complex after understanding the details of the issue so that the node polyfill had been detected instead of the dependency.

Thank you for your quick understanding of the actual problem and the fix. Great work, thank you :)

**Pull request merged**